src/features/views/hooks/useFolder.ts

```diff
@@ -0,0 +1,107 @@
+import deleteViewFolder from '../rpc/deleteFolder';
+import useViewTree from './useViewTree';
+import { ZetkinViewFolder } from '../components/types';
+import {
+  folderCreate,
+  folderCreated,
+  folderDeleted,
+  folderUpdate,
+  folderUpdated,
+} from '../store';
+import {
+  FutureBase,
+  IFuture,
+  PromiseFuture,
+  ResolvedFuture,
+} from 'core/caching/futures';
+import { useApiClient, useAppDispatch, useAppSelector } from 'core/hooks';
+
+type ZetkinViewFolderPostBody = {
+  parent_id?: number;
+  title: string;
+};
+
+type ZetkinViewFolderUpdateBody = Partial<
+  Omit<ZetkinViewFolder, 'id' | 'parent'>
+> & {
+  parent_id?: number | null;
+};
+
+interface UseFolderReturn {
+  createFolder: (title: string, folderId?: number) => Promise<ZetkinViewFolder>;
+  deleteFolder: (folderId: number) => void;
+  folderFuture: IFuture<ZetkinViewFolder>;
+  recentlyCreatedFolder: ZetkinViewFolder | null;
+  updateFolder: (
+    orgId: number,
+    folderId: number,
+    data: ZetkinViewFolderUpdateBody
+  ) => PromiseFuture<ZetkinViewFolder>;
```

Comment:
This hook is mixing data (`folderFuture`), mutations of an item (`deleteFolder`) and creation functions which mutate a collection (`createFolder`). I think they should be separate hooks.

This is not critical. If it works, it's fine. In other scenarios there are problems with using them together like this (e.g. unnecessary loading of data) but I don't think that's a big problem here.

Reply:
Not doing anything about this.